src/Akka.Persistence.Redis/Query/Stages/AllEventsSource.cs

```diff
@@ -0,0 +1,343 @@
+
+                            switch (_state)
+                            {
+                                case State.Idle:
+                                    // do nothing, no query is running and no client request was performed
+                                    break;
+                                case State.Querying:
+                                    _state = State.NotifiedWhenQuerying;
+                                    break;
+                                case State.NotifiedWhenQuerying:
+                                    // do nothing we already know that some new events may exist
+                                    break;
+                                case State.WaitingForNotification:
+                                    _state = State.Idle;
+                                    Query();
+                                    break;
+                            }
+                        }
+                        else
+                        {
+                            Log.Debug($"Message from unexpected channel: {data.channel}");
+                        }
+                    });
+
+                    _subscription = _redis.GetSubscriber();
+                    _subscription.Subscribe(_journalHelper.GetEventsChannel(), (channel, value) =>
+                    {
+                        messageCallback.Invoke((channel, value));
+                    });
+                }
+                else
+                {
+                    // start by first querying the current length of tag events
+                    // for the given tag
+                    // stream will stop once this has been delivered
+                    _state = State.Initializing;
+
+                    var initCallback = GetAsyncCallback<long>(len =>
+                    {
+                        _maxOffset = len;
+                        switch (_state)
+                        {
+                            case State.QueryWhenInitializing:
+                                // during initialization, downstream asked for an element,
+                                // let’s query elements
+                                _state = State.Idle;
+                                Query();
+                                break;
+                            case State.Initializing:
+                                // no request from downstream, just go idle
+                                _state = State.Idle;
+                                break;
+                            default:
+                                Log.Error($"Unexpected source state when initializing: {_state}");
+                                FailStage(new IllegalStateException($"Unexpected source state when initializing: {_state}"));
+                                break;
+                        }
+                    });
+
+                    _redis.GetDatabase(_database).ListLengthAsync(_journalHelper.GetEventsKey()).ContinueWith(task =>
```

Comment:
You are calling an async method, but I don't see any await on it. Is this intentional? At what point is the Task being executed?

Reply:
Yes, because `public override void PreStart()` is not async and `ContinueWith` helps us to handle any exception that might be thrown without `await`ing

Reply:
Please brush up your async/await knowledge.
`ContinueWith` returns a `Task`, if that `Task` is not awaited, it would not be executed.

Reply:
What will be the best approach here? Please enlighten me.

Reply:
To make an async `Task` perform work synchronously, you have to call `Task.Wait()` or if you want to capture the result, you have to capture `Task.Result`

Reply:
Doesn't that have the potential to `deadlock`? 

I have experienced it a lot and won't take that route. 

Everything goes fine first day, week, month, year and then suddenly, few seconds after, you get calls from outer space that the Earth is static - no movement(neither rotating nor revolving).

For `ListLengthAsync` and others alike, we are simply doing the job in the background. We handle the Task returned with `ContinueWith`.

Hope you do have a space in your heart for a "stubborn" teammate? cause am sticking with the current approach at my end ;)

Reply:
I have problem with the approach, but are you sure that the code was executed? Can you provide proof that it did?
A simple screenshot with a `Console.WriteLine()` would suffice.

Reply:
![image](https://user-images.githubusercontent.com/8049896/102904526-230dce80-4472-11eb-99e8-44a293e224a7.png)

Reply:
![image](https://user-images.githubusercontent.com/8049896/102904571-31f48100-4472-11eb-97af-3adc312483ac.png)